src/ErrorHandler.php

```diff
@@ -265,7 +270,24 @@ public function addExceptionHandlerListener(callable $listener): void
      */
     private function handleError(int $level, string $message, string $file, int $line, ?array $errcontext = []): bool
     {
-        if (0 === (error_reporting() & $level)) {
+        // The value `error_reporting()` returns when an error is captured that was silenced with the `@` operator was changed starting with PHP 8.
+        if (\PHP_MAJOR_VERSION >= 8) {
+            // On PHP 8 and above when an error is silenced with the `@` operator `error_reporting()` will be set to a list of error types
+            // that are now considered fatal and cannot be silenced anymore we subtract that list of the value `error_reporting()` reports
+            // and if we end up with `0` we know the error was silenced
+            $errorWasSilencedByUser = 0 === (error_reporting() & ~self::PHP8_ERROR_TYPES_THAT_CANNOT_BE_SILENCED);
+
+            // However this will mean we will erroneously mark errors that cannot be silenced as silenced so if the error
+            // level is on the list of error types that cannot be silenced we make sure the error is not marked as silent
+            if ($level === (self::PHP8_ERROR_TYPES_THAT_CANNOT_BE_SILENCED & $level)) {
+                $errorWasSilencedByUser = false;
+            }
+        } else {
+            // On PHP 7 and below when an error is silenced with the `@` operator `error_reporting()` will always return `0`
+            $errorWasSilencedByUser = 0 === error_reporting();
```

Comment:
Why this is different from before? It was:
```suggestion
            $errorWasSilencedByUser = 0 === error_reporting() & $level;
```

Reply:
No. That is exactly were the bug came from :) 

You can see that that does not work: https://3v4l.org/nct1E

Neither on PHP 7 or 8, but on PHP 7 and below we can safely assume it's always a literal 0.

Reply:
Holy crap this is really messy! Thanks for solving it.